File: HeartForCharity/HeartForCharity.Services/Database/CampaignMedia.cs
using HeartForCharity.Model.Enums;
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace HeartForCharity.Services.Database
{
    public class CampaignMedia
    {
        [Key]
        public int CampaignMediaId { get; set; }

        [Required]
        [ForeignKey(nameof(Campaign))]
        public int CampaignId { get; set; }

        [Required]
        [MaxLength(500)]
        public string Url { get; set; } = null!;

        public MediaType MediaType { get; set; } = MediaType.Image;
        public bool IsCover { get; set; } = false;

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public virtual Campaign Campaign { get; set; } = null!;
    }


}

File: HeartForCharity/UI/heartforcharity_desktop/lib/model/responses/campaign_media.dart
class CampaignMedia {
  final int campaignMediaId;
  final String? url;
  final String? mediaType;
  final bool isCover;

  CampaignMedia({
    this.campaignMediaId = 0,
    this.url,
    this.mediaType,
    this.isCover = false,
  });

  factory CampaignMedia.fromJson(Map<String, dynamic> json) => CampaignMedia(
        campaignMediaId: json['campaignMediaId'] ?? 0,
        url: json['url'],
        mediaType: json['mediaType'],
        isCover: json['isCover'] ?? false,
      );

  Map<String, dynamic> toJson() => {
        'campaignMediaId': campaignMediaId,
        'url': url,
        'mediaType': mediaType,
        'isCover': isCover,
      };
}
